src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
The idea to start with children is valid, however, please consider some updates:
1. You basically pick all people with mothers. That makes little sense as most people have known mothers so the request will be ineffective, as when calculating the age diff you will search for the 
2. Try changing the code to have a single `filter` call

However, the solution will work, if you will pair all children with their mothers:

```
people.map(person => {child: person, mother: [find the mother]}).filter([remove people without known mother and filter men, if `onlyWithSon` present])
```